### ISSUE TITLE
Regression in 3.4 with internal linking and go back in history

### ISSUE THREAD

**Issue by GeorgeKerscher** [edrlab/thorium-reader#3560]
Description: We have discovered an issue in Thorium 3.4 on Windows 11. Using the fundamental non-visual test book and the Go back in history test book both available from epubtest.org,  following links and using go back (ctrl + backspace) are intermittently not working. 

This has been reproduced on four computers by George, Prashant, Sanjog, and Richard.
This does not happen all the time, but it is probably 50% of the time. This is with Jaws, NVDA, and Narrator.,

When following a link to a footnote or a link to another place in the title, we are taken to a frame with no content. It says:
Go backward
Go forward
Jump back in navigation historyJump forward in navigation history

When I press ctrl + backspace, I am taken to the beginning of the chapter or to the toolbar and the chapter that initiated the link is present.

When I activate the footnote link and I am in the empty frame, when I press where am I (ctrl +shift +I) it says:
Progression
4/5 (Non-Visual Reading) [22%]
h1Non-Visual Reading Tests/h2reading-420 Footnote Reading

The behavior is slightly different in Jaws, NVDA, and Narrator, but all display this intermittent behavior.

Let me know if I should submit a video

**Comment by danielweck** [comment]
@gautierchomel provided a screen recording that showcases the behaviour. it looks like the screen reader detection and / or the user activation is not operational, causing the shifts in reading location.
I need to reproduce at my end so I can troubleshoot the system events.

**Comment by GeorgeKerscher** [comment]
Glad you got to see it. Let me know if there is anything I can do to help.

Best
George

**Comment by danielweck** [comment]
I am unable to reproduce this problem on MacOS with VoiceOver (latest build of Thorium Desktop reader which will ship as version 3.5.0). It is as if you are using the application with a screen reader but without first ticking the box in the application settings to activate the screen reader enhancement. Could you please clarify? Feel free to provide a video recording but note that the focus ring of screen readers sometimes is incorrectly located in the scrolling viewport of the web document, so it is hard for me to visually track what your aural experience is.

**Comment by GeorgeKerscher** [comment]
We have been mainly testing on Windows 11 computers. Do you want a video from my Windows machines? Or is that not useful.

**Comment by danielweck** [comment]
Thank you George, a screen recording could help, for example I might see that the table of contents or some other GUI control somehow interferes with keyboard focus. Ideally you would be recording the bug while using a build of Thorium Desktop reader based on the latest code, not with version `3.4.0`. The Windows build is now ready, the unsigned installer can be downloaded from the usual link for Windows / Intel x64: https://github.com/edrlab/thorium-reader/releases/tag/latest-windows-intel

**Comment by danielweck** [comment]
New issue filed: https://github.com/edrlab/thorium-reader/issues/3777
(after the bugfixes https://github.com/edrlab/thorium-reader/commit/2695d6b3a8ca234f9fe49938e4079f2ab5136e71 )
